gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,244 @@
+    return self._variant.reference_bases
+
+  def get_names(self):
+    return self._variant.names
+
+  def get_quality(self):
+    return self._variant.quality
+
+  def get_filters(self):
+    return self._variant.filters
+
+  def get_calls(self):
+    return self._variant.calls
+
+  def get_non_alt_infos(self):
+    return self._non_alt_infos
+
+  def get_alternate_data_list(self):
+    if self._single_alternate_datas == None:
+      raise ValueError(
+          'This class should be constructed using the factory function.')
+    return self._single_alternate_datas
+
+  # Mutation functions. These are intended to be used by the factory class,
+  # hence they are not "public".
+
+  def _add_per_alt_info(self, field_name, variant_info_data):
+    if len(variant_info_data) != len(self._single_alternate_datas):
+      raise ValueError(
+          'Per alternate INFO field {} does not have same cardinality as '
+          ' number of alternates: {} vs {}'.format(
+              field_name, len(variant_info_data),
+              len(self._single_alternate_datas)))
+    for alt_index, info in enumerate(variant_info_data):
+      self._single_alternate_datas[alt_index]._add_single_info(field_name, info)
+
+  def _add_annotation(self, field_name, data, header_desc):
+    """Adds an annotation INFO field based on the format in the header.
+
+    Args:
+      field_name (str): The name of the annotation field.
+      data(List[str]): The data part of the field separated on comma.
+      header_desc (str): The description of the field in the header used for
+        extracting the name of different annotations in `data`.
+    """
+    annotation_names = extract_annotation_names(header_desc)
+    for annotation_str in data:
+      annotations = extract_annotation_list_with_alt(annotation_str)
+      if len(annotation_names) != len(annotations) - 1:
+        raise ValueError('Expected {} annotations, got {}'.format(
+            len(annotation_names), len(annotations) - 1))
+      found_alt = False
+      # This assumes that number of alternate bases and annotation segments
+      # are not too big. If this assumption is not true, we should replace the
+      # following loop with a hash table search and avoid the quadratic time.
+      for alt in self._single_alternate_datas:
+        if alt.get_alternate_bases() == annotations[0]:
+          found_alt = True
+          alt._add_annotations(field_name, annotations[1:], annotation_names)
+          break
+      # TODO(bashir2): Currently we only check exact matches of alternate bases
+      # which is not enough. We should implement the whole standard for finding
+      # alternate bases for an annotation list.
+      if not found_alt:
+        logging.warning('Could not find matching alternate bases for {} in '
+                        'annotation filed {} with content {}'. format(
+            annotations[0], field_name, annotation_str))
+
+
+class SingleAlternateData(object):
```

Comment:
Consider renaming this class to be more descriptive. Perhaps VariantAlternateBase? (similar to VariantCall). I think this should also just be a simple storage class (similar to a proto).